include/mbedtls/mbedtls_config.h

```diff
@@ -0,0 +1,67 @@
+/**
+ * \file mbedtls_config.h
+ *
+ * \brief Configuration options (includable header)
+ *
+ *  Include this file if you need to depend on the
+ *  configuration options defined in mbedtls_config_defines.h
+ */
+ /*
+  *  Copyright The Mbed TLS Contributors
+  *  SPDX-License-Identifier: Apache-2.0
+  *
+  *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+  *  not use this file except in compliance with the License.
+  *  You may obtain a copy of the License at
+  *
+  *  http://www.apache.org/licenses/LICENSE-2.0
+  *
+  *  Unless required by applicable law or agreed to in writing, software
+  *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+  *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  *  See the License for the specific language governing permissions and
+  *  limitations under the License.
+  */
+
+#ifndef MBEDTLS_MBEDTLS_CONFIG_H
+#define MBEDTLS_MBEDTLS_CONFIG_H
+
+#if defined(_MSC_VER) && !defined(_CRT_SECURE_NO_DEPRECATE)
+#define _CRT_SECURE_NO_DEPRECATE 1
+#endif
+
+#if !defined(MBEDTLS_CONFIG_FILE)
+#include "mbedtls/mbedtls_config_defines.h"
+#else
+#include MBEDTLS_CONFIG_FILE
+#endif
+
+#if !defined(MBEDTLS_CONFIG_VERSION) || \
+    MBEDTLS_CONFIG_VERSION != 1
```

Comment:
I think it's confusing to have version 1 be the version introduced in Mbed TLS 3.0. Should we make this version 3? Or version 0x030000? Or version 202106?

Reply:
What about for the version of the configuration to just follow the version of the library ? That way we don't have two different versions for the config and the library.
At each release we update in the editable config.h the config version to that of the library version.
Then for a library version  of major version M >=  3, only configuration versions MBEDTLS_CONFIG_VERSION lower than the library version and with the same major version are acceptable. If MBEDTLS_CONFIG_VERSION is strictly lower than the library version MBEDTLS_VERSION_NUMBER the resulting configuration is as if the library version was equal to MBEDTLS_CONFIG_VERSION.